astropy/time/core.py

```diff
@@ -437,6 +438,9 @@ def precision(self, val):
             raise ValueError('precision attribute must be an int between '
                              '0 and 9')
         self._precision = val
+        if hasattr(self, '_time'):
```

Comment:
Have to think this one through better, but would it be an option to let `precision` refer directly to `_time.precision`? Or is it important that it can be set while there is no `_time` associated (yet)?